pkg/client/restclient/request.go

```diff
@@ -910,12 +910,26 @@ func (r *Request) transformResponse(resp *http.Response, req *http.Request) Resu
 		return Result{err: errors.FromObject(status)}
 	}
 
-	// TODO: Check ContentType.
+	contentType := resp.Header.Get("Content-Type")
+	var decoder runtime.Decoder
+	if contentType == r.content.ContentType {
```

Comment:
I'm not sure this is safe in all cases - the string content-type may only be partially equivalent.  I think we have to do a slightly more invasive check.

At a minimum, in your else block you need to call `mime.Parse(contentType)` and pass both parts to renegotiate.